src/Idris/Parser.idr

```diff
@@ -793,15 +793,27 @@ mutual
 
   record_ : OriginDesc -> IndentInfo -> Rule PTerm
   record_ fname indents
-      = do b <- bounds (do kw <- option False
-                                 (decoratedKeyword fname "record"
-                                   $> True) -- TODO deprecated
-                           decoratedSymbol fname "{"
-                           commit
-                           fs <- sepBy1 (decoratedSymbol fname ",") (field kw fname indents)
-                           decoratedSymbol fname "}"
-                           pure $ forget fs)
+      = do
+           b <- (
+               withWarning oldSyntaxWarning (
+                 bounds (do
+                   decoratedKeyword fname "record"
+                   body True
+                 ))
+             <|>
+               bounds (body False))
            pure (PUpdate (boundToFC fname b) b.val)
+    where
+      oldSyntaxWarning : String
+      oldSyntaxWarning = "DEPRECATED: old record update syntax. Use \"{ f := v } p\" instead of \"{ f = v } p\""
```

Comment:
```suggestion
      oldSyntaxWarning = "DEPRECATED: old record update syntax. Use \"{ f := v } p\" instead of \"record { f = v } p\""
```

Reply:
Maybe we should use a better? message
![image](https://user-images.githubusercontent.com/16068886/146195921-b828cbb4-faa4-4934-b923-8541f37aabaf.png)

Reply:
That might be a better message. The thing that might be most confusing to a user is the subtle difference between an equals sign and a colon-equals or the fact that the new syntax suggestion could be interpreted as simply not showing the "record" keyword again instead of indicating that the keyword should not be used. At the end of the day I think either of the messages you've suggested would be good.

Reply:
I'll stick to the first message then